common/app/model/ShareLinks.scala

```diff
@@ -93,7 +93,7 @@ object ShareLinks {
     lazy val facebookParams = List(
       Some("app_id" -> facebookAppId),
       Some("href" -> encodedHref),
-      quote.map(q => "quote" -> q),
+      quote.map(q => "quote" -> q.urlEncoded),
       mediaPath.map(path => "picture" -> path.urlEncoded)
```

Comment:
so actually it's not to do with url encoding at all, actually { and } are special substitution characters for the uribuilder.  So those need to be escaped (not clear how though), not the " etc.  The UriBuilder will do the rest.  The escaping of `{` and `}` should be done in the `appendQueryParams` method, not in every piece of code that calls it.